Sources/MySQL/Database.swift

```diff
@@ -64,6 +67,21 @@ public final class Database {
         self.socket = socket
         self.flag = flag
         self.encoding = encoding
+        
+        // Clamp connection pool size to range of 1-32
+        let poolSize = pool < 1 ? 1 :
+            pool > 32 ? 32 :
+            pool
+        
+        self.connectionPool = [Connection]()
+        for _ in 0..<poolSize {
+            do {
+                let conn = try self.makeConnection()
```

Comment:
@tannernelson Good point. I just ran some tests, and creating a pool of 32 connections took 0.01 seconds with the MySQL server running on localhost, and 5 seconds for a remote server. So yeah it looks like it can get slow for remote databases.

I'll look into creating them on the fly.